### DISCUSSION TITLE
How to handle errors from a CMS?

### DISCUSSION THREAD

**Discussion by kwiat1990** [TheRealFlyingCoder/remix-auth-socials#16] [Q&A]
Hey, I have followed docs from `remix-auth` and `remix-auth-socials` to authenticate my users with Google and Facebook. Normally, if a user authenticates for the very first time a new account will be created in my CMS for that user. Following requests will simply return his existing account. My case: if an existing user tries to login with another social provider, my CMS returns an error because it has already a user with his e-mail and linked to other provider. What I want to achieve in such case is to redirect such user to the login page and show him error message from the CMS, so the he knows that he needs to use another provider.

**Comment by TheRealFlyingCoder** [comment]
Hey sorry this is like a year later,

In this scenario you are just adding a default check in the remix-auth function for each social provider.... If the user exists and didn't use this social oauth, throw specific error message and redirect to login.

**Reply by kwiat1990** [reply]
Thanks for the answer but I have a problem implementing this feature as I don't have a dedicated login page. Instead, I have a login component embedded in a comment system displayed on a blog post page, which displays your current profile or let a user use Google or Facebook to login. So that in either case - a successful or failed one - he will be redirected to the original page. And in this moment, I would like to read an error from the session if it exists. After successful login the previously logged error should be erased from the session. I thought that:
```ts
"__flash_auth:error__": {
    "message": "Email is already taken."
  }
```
will be destroyed after successful login but instead it persists in the user's session.

**Reply by kwiat1990** [reply]
@TheRealFlyingCoder, it seems that authorization errors thrown in `auth.server.ts` doesn't render Error Boundary. I have a global one in `root.ts` and also one in `/auth/$provider.callback.tsx` but none of them will be rendered in case of thrown error. Any idea why is that?

**Reply by TheRealFlyingCoder** [reply]
> @TheRealFlyingCoder, it seems that authorization errors thrown in `auth.server.ts` doesn't render Error Boundary. I have a global one in `root.ts` and also one in `/auth/$provider.callback.tsx` but none of them will be rendered in case of thrown error. Any idea why is that?

Hey, it depends on how you've set remix up, to be honest this isn't really a remix-auth-socials thing, it's more just general Remix / Auth stuff.

I'd just read the docs and make sure you understand what's supposed to happen, remix-auth as well, and use the discord help channel if you need more info

**Reply by kwiat1990** [reply]
I managed to solve issue in my code by reading Remix docs regarding how to handle session flash. It wasn’t clear to me that after reading flash message, one should `commitSession` afterwards.
